api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -343,6 +351,17 @@ func (m *AzureManagedMachinePool) validateEnableNodePublicIP() error {
 	return nil
 }
 
+func (m *AzureManagedMachinePool) validateSubnetName() error {
+	if m.Spec.SubnetName != "" {
+		subnetRegex := `^[-\w\._]+$`
```

Comment:
I maybe wrong, but this regex seems to be validating `-_-_-_-_`. Do you think we could make this more strict on accepting well-formed strings? Or is `-_-_-_-_` an acceptable subnet name? 
What does Azure Docs say about the subnet name?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/016209aa3b34393e8c23457f5afdf36db4fb5559/api/v1beta1/azurecluster_validation.go#L43

Copied it from here.

Reply:
@nawazkh  good catch :)